libr/core/rvc.c

```diff
@@ -355,46 +355,65 @@ static bool rm_empty_dir(const char *rp) {
 	return true;
 }
 
-static RList *repo_files(const char *rp) {
-	RList *files = r_file_lsrf (rp);
-	if (!files) {
-		return NULL;
+//This is why anonymous functions are cool
+static bool dr(RList *dst, const char *dir) {
+	char *name;
+	RListIter *iter;
+	bool ret = true;
+	RList *files = r_sys_dir (dir);
+	if (!r_list_empty (dst)) {
+		char *vcp = r_str_newf ("%s" R_SYS_DIR ".rvc", dir);
+		if (!vcp) {
+			return false;
+		}
+		if (r_file_is_directory (vcp)) {
+			return true;
+		}
 	}
-	char *repo_meta = r_str_newf ("%s" R_SYS_DIR ".rvc", rp);
-	if (!repo_meta) {
-		r_list_free (files);
-		return NULL;
+	if (!files) {
+		return false;
 	}
-	size_t repo_meta_len = r_str_len_utf8 (repo_meta);
-	RListIter *iter;
-	char *path;
-	RList *ret = r_list_new ();
-	r_list_foreach (files, iter, path) {
-		if (!r_str_cmp (repo_meta, path, repo_meta_len)) {
+	r_list_foreach (files, iter, name) {
+		char *path;
+		if (!strcmp (name, "..") || !strcmp (name, ".")) {
 			continue;
 		}
-		char *absp = r_file_abspath (path);
-		if (!absp) {
-			r_list_free (ret);
-			ret = NULL;
+		if (!strcmp (name, ".rvc")) {
+			continue;
+		}
+		path = r_str_newf ("%s" R_SYS_DIR "%s", dir, name);
+		if (!path) {
+			ret = false;
 			break;
 		}
-		if (r_file_is_directory (absp)) {
-			free (absp);
+		if (r_file_is_directory (path)) {
+			if (!dr (dst, path)) {
+				ret = false;
+				break;
+			}
+			free (path);
 			continue;
 		}
-		if (!r_list_append (ret, absp)) {
-			r_list_free (ret);
-			ret = NULL;
-			free (absp);
+		if (!r_list_append (dst, path)) {
+			ret = false;
+			free (path);
 			break;
 		}
-
 	}
-	free (repo_meta);
 	r_list_free (files);
 	return ret;
+}
 
+static RList *repo_files(const char *dir) {
+	RList *ret = r_list_new ();
+	if (!ret) {
```

Comment:
Remove this

Reply:
Remove repo_files? but if I did we would habe to alloc everytime we list the files. That would be 9 lines per repo_file.